sdk/core/Azure.Core/tests/RetryPolicyTestBase.cs

```diff
@@ -384,6 +385,219 @@ public async Task RetryingEmitsEventSourceEvent()
             AssertRetryEvent(listener, request, 2);
         }
 
+        [Test]
```

Comment:
Could we create a test that exercises the thread-safe scenario where the index is not incremented because the host value is not the same as the current fallback value?